Prevent NetworkManager to change /etc/resolv.conf

Sunnatillo opened the pull request:
This PR disables NetworKManager resolv.conf file changes. Instead adds dns addresses manually for centos images.

tuminoid commented on ci/scripts/image_scripts/configure_network_centos.sh:
```diff
@@ -0,0 +1,16 @@
+# Specify DNS servers for systemd-resolved.
+cat <<EOF | sudo tee /etc/systemd/resolved.conf
+[Resolve]
+DNS=8.8.8.8 8.8.4.4
```
We're using Cloudflare's 1.1.1.1 and 1.0.0.1 as primary DNS on Ubuntu, any reason why we'd default to Google on Centos?

Sunnatillo replied:
There is no specific reason behind. As external dns I specified Google's dns addresses as this is commonly used. For fallback, wanted to use different dns addresses. 
Should we use the same?

tuminoid replied:
I see you split the difference and changed ubuntu and centos to match, and use both. Nice.

Sunnatillo commented:
/test-integration-metal3-dev-tools-centos

Sunnatillo commented:
/test-integration-metal3-dev-tools-centos

Sunnatillo commented:
/test-integration-metal3-dev-tools-centos

kashifest commented:
We had similar kind of configuration which we removed here https://github.com/Nordix/metal3-dev-tools/pull/674/ so I am trying to understand why we removed it and what problems would we face if we reintroduce it @Rozzii

Sunnatillo commented:
> We had similar kind of configuration which we removed here #674 so I am trying to understand why we removed it and what problems would we face if we reintroduce it @Rozzii

We previously disabled whole cloud init. and it Prevented us to set 

> We had similar kind of configuration which we removed here #674 so I am trying to understand why we removed it and what problems would we face if we reintroduce it @Rozzii

We previously disable whole cloud init and it prevented us to use IPAN based IP addresses. @Rozzii removed the changes with his changes. This PR only disabling setting dns address by NetworkManager.

lentzi90 commented:
Wondering the same as @kashifest . We have been going back and forth with this. I don't think writing directly to `resolv.conf` is the right way to go. That will end up with having to make the file immutable to prevent cloud-init from overwriting it and then facing cloud-init errors instead.
https://github.com/Nordix/metal3-dev-tools/pulls?q=is%3Apr+dns+networkmanager+is%3Aclosed

Why don't we go back to something like this instead:
https://github.com/Nordix/metal3-dev-tools/blob/c22b7c85c69cc5d775bcccabd7141749d083bf67/ci/scripts/image_scripts/configure_network_centos.sh

lentzi90 commented:
Another idea is also to configure this in Cleura. We can set the nameservers to use directly on the subnet there.

Rozzii commented:
/hold

Sunnatillo commented:
/test-integration-metal3-dev-tools-ubuntu

Rozzii commented:
We had a discussion and for now this will get a hold as my opinion is that DNS server should be specified via Metal3DataTemplate.

Sunnatillo commented:
/close

As we discussed it is better to set dns address through metal3datatemplate.
